src/back/TicketManager/tests/Core/TicketManager.Core.DomainTests/Events/EventBuilderTests.cs

```diff
@@ -0,0 +1,287 @@
+            .WithLocation(location)
+            .WithDate(DateTime.UtcNow.AddDays(-1))
+            .WithSectors(sectors)
+            .Build();
+
+        func.Should().Throw<ArgumentException>();
+    }
+    
+    [Fact]
+    public void WhenSectorsAreNotProvided_ItShouldThrowArgumentException()
+    {
+        var func = () => new EventBuilder()
+            .WithGeneratedId()
+            .WithOrganizerId(organizerId)
+            .WithName(name)
+            .WithDescription(description)
+            .WithLocation(location)
+            .WithDate(date)
+            .Build();
+
+        func.Should().Throw<ArgumentException>();
+    }
+    
+    [Fact]
+    public void WhenSectorsWithTheSameNameAreProvided_ItShouldThrowArgumentException()
+    {
+        var invalidSectors = new List<EventBuilder.SectorData>
+        {
+            new("name", 1, 1, 1),
+            new("name", 2, 2, 2),
+            new("name", 3, 3, 3),
+        };
+        
+        var func = () => new EventBuilder()
+            .WithGeneratedId()
+            .WithOrganizerId(organizerId)
+            .WithName(name)
+            .WithDescription(description)
+            .WithLocation(location)
+            .WithDate(date)
+            .WithSectors(invalidSectors)
+            .Build();
+
+        func.Should().Throw<ArgumentException>();
+    }
+    
+    [Fact]
+    public void WhenSectorWithNegativePriceIsProvided_ItShouldThrowArgumentException()
+    {
+        var invalidSectors = new List<EventBuilder.SectorData>
+        {
+            new("name1", -1, 1, 1),
+            new("name2", 2, 2, 2),
+            new("name3", 3, 3, 3),
+        };
+        
+        var func = () => new EventBuilder()
+            .WithGeneratedId()
+            .WithOrganizerId(organizerId)
+            .WithName(name)
+            .WithDescription(description)
+            .WithLocation(location)
+            .WithDate(date)
+            .WithSectors(invalidSectors)
+            .Build();
+
+        func.Should().Throw<ArgumentException>();
+    }
+    
+    [Fact]
+    public void WhenSectorWithNegativeNumberOfRowsIsProvided_ItShouldThrowArgumentException()
+    {
+        var invalidSectors = new List<EventBuilder.SectorData>
+        {
+            new("name1", 1, 1, -1),
+            new("name2", 2, 2, 2),
+            new("name3", 3, 3, 3),
+        };
+        
+        var func = () => new EventBuilder()
+            .WithGeneratedId()
+            .WithOrganizerId(organizerId)
+            .WithName(name)
+            .WithDescription(description)
+            .WithLocation(location)
+            .WithDate(date)
+            .WithSectors(invalidSectors)
+            .Build();
+
+        func.Should().Throw<ArgumentException>();
+    }
+    
+    [Fact]
+    public void WhenSectorWithNegativeNumberOfColumnsIsProvided_ItShouldThrowArgumentException()
+    {
+        var invalidSectors = new List<EventBuilder.SectorData>
+        {
+            new("name1", 1, 1, -1),
```

Comment:
It should be `new("name", 1, -1, 1),` , you specified `row` to be tested in `column` test